documentation/writing_tests.md

```diff
@@ -14,6 +20,139 @@
+- `--full-trace`, which enables full stack traces, since Mocha by default shortens the stack traces.
+
+That command by itself is still missing two things: the actual test files and a way to generate code coverage.
+These two missing pieces can vary depending on what we're trying to test and how we're trying to debug the tests.
+
+Code coverage can be added by placing `nyc` at the beginning of the line.
+Keep in mind that `nyc` will **obscure the stack traces**, so it's preferable to
+make separate `package.json` scripts, one for automated testing through CI, with `nyc`,
+and another one for developers running tests, without `nyc`.
+
+Then we have to point mocha to our test files. If you're **not** using `nyc`, you can point to the
+bundled test file (bundled with rollup, which we will see later), typically at `dist-test/index.node.js`.
+If you are using `nyc`, point mocha to the files built by the TypeScript compiler, normally at `dist-esm/test/*.test.js`.
+
+With these considerations in mind, you can see how your package.json scripts should look at the section:
+[package.json scripts](#package-json-scripts).
+
+#### Handling timeouts
+
+All of the tests in the Azure SDK repository should have a timeout. In the previous section, we have discussed how this timeout is defined for mocha, though the parameter `--timeout` with an additional milliseconds amount, which is sent during the invocation of mocha. Now we will define how to know how many milliseconds to pass to this parameter.
+
+New projects inside of this repository can take a safe guess based on how much the tests on average take locally. **Existing projects** should instead...
+
+    TODO: ASK ENGINEERING HOW TO GET THE MAX TIME PER TESTS
+
+#### On the usage of before, beforeEach, after and afterEach
+
+Mocha's `before` and `beforeEach` are methods that allow you to specify functions that should be executed either before all of the tests run or before each test runs, respectively. Similarly, `after` and `afterEach` exist so that developers can specify functions that should be executed after all tests run or after each test runs, respectively.
+
+We recommend using `beforeEach` rather than `before`, just as much as we recommend using `afterEach` rather than `after`. The idea is that each one of your test cases should not depend on a state that is shared with other tests. Use `beforeEach` to execute tasks that will prepare the resources needed for each test to run cleanly, and `afterEach` to tear down or clean those settings before the next test runs.
+
+We typically use `beforeEach` and `afterEach` to set up and tear down our test recorder. You can learn more about it in the section: [The Recorder](#the-recorder).
+
+#### Other general recommendations
+
+Mocha has many interesting features. Here's a list of our general recommendations on how to use some of these features:
```

Comment:
I would greatly appreciate a code snippet showing positive examples of how to write tests that are cleanly copy-pasteable

Reply:
I'm planning to have a whole section where I get into the details of how to write test cases 🌞 The title of that section is here: https://github.com/Azure/azure-sdk-for-js/pull/7784/files#diff-a43d448b275343b7d964b6e68f7cabf4R19

This PR is intended to have only recommendations about the tools we're using.

Having that in mind, would yous till like code snippets here? What kind of snippets? I can pick up and expand once I have a better idea of how to improve this part 🌞

Reply:
a link to the other bit that talks about them would be fine